linode/manager
# fix: [M3-7880] - Properly reset rebuild modal state

## Description 📝
This PR fixes a bug where the "mode" state of the rebuild dialog wasn't reset when mounted. This isn't as bad a bug as initially thought, just a potential awkward experience for the user.

## Changes  🔄
- Reset state when closing modal

## Target release date 🗓️
3/18/2024

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/130582365/0e85eb3f-266d-4a49-9015-59d8cddf6652" /> | <video src="https://github.com/linode/manager/assets/130582365/5e2fd25c-849f-4dfb-8a7c-64a5eb1aae94" /> |

## How to test 🧪

### Reproduction steps
- Navigate to linode list, click on "Rebuild" action menu item for **Linode A**
- Close modal, click on "Rebuild" action menu item for **Linode A**
  - Notice the previous mode selection

- Navigate to linode list, click on "Rebuild" action menu item for **Linode A**
- Close modal, click on "Rebuild" action menu item for **Linode B**
  - Notice the previous mode selection

### Verification steps
- Navigate to linode list, click on "Rebuild" action menu item for **Linode A**
- Close modal, click on "Rebuild" action menu item for **Linode A**
  - Notice the mode selection was reset

- Navigate to linode list, click on "Rebuild" action menu item for **Linode A**
- Close modal, click on "Rebuild" action menu item for **Linode B**
  - Notice the mode selection was reset
 
## As an Author I have considered 🤔

*Check all that apply*
